### PR TITLE
refactor: add documentation, tests and improvements

### PR DESCRIPTION
**Pull request by protiumx**
In this PR:
- I have added documentation explaining the implementation of `HBExporter` and the export process itself.
- I documented improvements in code as `TODO`s and added `docs/improvements.md` with further notes.
- I have added `jest` and `jest-ts` for unit testing. This PR achieves 100% of coverage

Note: `index.ts` was left out of the coverage as this files does not form part of the `exporter` implementation.

### CODE REVIEW

File: test/unit/exporter.spec.ts [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+      (PermissionMock.CheckPermissions as jest.Mock).mockResolvedValueOnce(
+        false
+      );
+      const deps: Partial<HBExporterDependencies> = {
+        allowedPermission: "export",
+        permissionsService: PermissionMock,
+      };
+      const user: User = { id: "user", permissions: [] };
+      const exporter = HBExporter(deps as HBExporterDependencies);
+      const data = new PassThrough();
```
Comment (christianmellor) [inline test/unit/exporter.spec.ts:68]: Is there a reason why you use `PassThrough` here?

File: test/unit/exporter.spec.ts [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  readable.push(null);
+
+  return readable;
+});
+
+const exportId = "0000-0000";
+
+describe("exporter", () => {
+  describe("StartExport", () => {
+    it("should rethrow any error", async () => {
```
Comment (christianmellor) [inline test/unit/exporter.spec.ts:38]: Is there a more appropriate name for this test?

File: test/unit/exporter.spec.ts [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+      const ret = await exporter.StartExport(user, data);
+      expect(ret).toEqual(expectedStatus);
+      expect(CacheMock.SET).toHaveBeenCalledWith(
+        exportId,
+        JSON.stringify(expectedStatus),
+        expect.any(Function)
+      );
+    });
+
+    it("should append chunks of data and update export status to PENDING", async () => {
```
Comment (christianmellor) [inline test/unit/exporter.spec.ts:106]: I like this test, great stuff! 🎉

**Review by christianmellor — COMMENTED** [review]
Looks fantastic! One final question, in the tests you often use the `Partial` generic function and then the type assertion `deps as HBExporterDependencies`. Is there any reason to declare `const deps: Partial<HBExporterDependencies>` if you then force a type assertion anyway? Can you think of any drawbacks of doing this?

Other than that, we look forward to discussing with you in the next meeting 🙌

File: test/unit/exporter.spec.ts (hunk 2 above)
Comment (protiumx) [reply test/unit/exporter.spec.ts]: Good catch. I have renamed this to match what we expect when handling any error.

**Pull request closed without merging**